date_module.js

```diff
@@ -0,0 +1,122 @@
+var myObj = ( function iife() {
+  var myDate = null;
+  var DAYS = [
+    'Sunday',
+    'Monday',
+    'Tuesday',
+    'Wednesday',
+    'Thursday',
+    'Friday',
+    'Saturday'
+  ];
+
+  var MONTHS = [
+    'January',
+    'February',
+    'March',
+    'April',
+    'May',
+    'June',
+    'July',
+    'August',
+    'September',
+    'October',
+    'November',
+    'December'
+  ];
+
+  function _initializeMyDate() {
+    myDate = new Date();
+  }
+  function setDate( date ) {
+    if (typeof date === 'number' || date instanceof Date) {
+      date = new Date(date);
```

Comment:
I think what you meant to do here is `myDate = new Date(date);` - by reassigning to the local date variable, you lose the effect of what this method is supposed to do, which is set the internal date for the module.